packages/remix-react/__tests__/integration/meta-test.tsx

```diff
@@ -0,0 +1,316 @@
+            },
+          },
+        ],
+      },
+    ]);
+
+    let { container } = render(<RemixStub />);
+
+    expect(getHtml(container)).toMatchInlineSnapshot(`
+      "<div>
+        <meta
+          charset=\\"utf-8\\"
+        />
+        <title>
+          Child title
+        </title>
+        <p>
+          Matches Meta
+        </p>
+      </div>"
+    `);
+  });
+
+  it("{ charSet } adds a <meta charset='utf-8' />", () => {
+    let RemixStub = unstable_createRemixStub([
+      {
+        path: "/",
+        meta: () => [{ charSet: "utf-8" }],
+        Component: Meta,
+      },
+    ]);
+
+    let { container } = render(<RemixStub />);
+
+    expect(getHtml(container)).toMatchInlineSnapshot(`
+      "<div>
+        <meta
+          charset=\\"utf-8\\"
+        />
+      </div>"
+    `);
+  });
+
+  it("{ title } adds a <title />", () => {
+    let RemixStub = unstable_createRemixStub([
+      {
+        path: "/",
+        meta: () => [{ title: "Document Title" }],
+        Component: Meta,
+      },
+    ]);
+
+    let { container } = render(<RemixStub />);
+
+    expect(getHtml(container)).toMatchInlineSnapshot(`
+      "<div>
+        <title>
+          Document Title
+        </title>
+      </div>"
+    `);
+  });
+
+  it("{ property: 'og:*', content: '*' } adds a <meta property='og:*' />", () => {
+    let RemixStub = unstable_createRemixStub([
+      {
+        path: "/",
+        meta: () => [
+          { property: "og:image", content: "https://picsum.photos/200/200" },
+          { property: "og:type", content: undefined },
+        ],
+        Component: Meta,
+      },
+    ]);
+
+    let { container } = render(<RemixStub />);
+    expect(getHtml(container)).toMatchInlineSnapshot(`
+      "<div>
+        <meta
+          content=\\"https://picsum.photos/200/200\\"
+          property=\\"og:image\\"
+        />
+        <meta
+          property=\\"og:type\\"
+        />
+      </div>"
+    `);
+  });
+
+  it("{ 'script:ld+json': {} } adds a <script type='application/ld+json' />", () => {
+    let jsonLd = {
+      "@context": "http://schema.org",
+      "@type": "Person",
+      name: "Sonny Day",
+      address: {
+        "@type": "PostalAddress",
+        streetAddress: "123 Sunset Cliffs Blvd",
+        addressLocality: "San Diego",
+        addressRegion: "CA",
+        postalCode: "92107",
+      },
+      email: ["sonnyday@fancymail.com", "surfergal@veryprofessional.org"],
+    };
+
+    let RemixStub = unstable_createRemixStub([
+      {
+        path: "/",
+        meta: () => [
+          {
+            "script:ld+json": jsonLd,
+          },
+        ],
+        Component: Meta,
+      },
+    ]);
+
+    let { container } = render(<RemixStub />);
+
+    // For some reason, prettyDOM strips the script tag (maybe because of
+    // dangerouslySetInnerHTML), so we just parse the HTML out into JSON and assert that way
+    let scriptTagContents =
+      container.querySelector("script")?.innerHTML || "{}";
```

Comment:
Should this test also ensure the correct `type` attribute is present?

```suggestion
      container.querySelector('script[type="application/ld+json"]')
        ?.innerHTML || "{}";
```